R/utilities-simulation.R

```diff
@@ -1,53 +1,146 @@
+      logFolder = logFolder
+    )
+  }
+  else if (numberOfCores > 1) {
+    logWorkflow(
+      message = "Starting parallel population simulation",
+      pathFolder = logFolder
+    )
+
+    simulationResultFileNames <- runParallelPopulationSimulation(
+      numberOfCores = numberOfCores,
+      structureSet = structureSet,
+      settings = settings,
+      logFolder = logFolder
+    )
+
+    simulationResult <- ospsuite::importResultsFromCSV(simulation = ospsuite::loadSimulation(structureSet$simulationSet$simulationFile),
+                                                       filePaths = simulationResultFileNames)
+    file.remove(simulationResultFileNames)
+
+    logWorkflow(
+      message = "Parallel population simulation completed.",
+      pathFolder = logFolder
+    )
+
+  }
+
+  return(simulationResult)
+
+}
+
+#' @title simulateModelOnCore
+#' @description Simulate model, either for an individual or for a given population.
+#' @param structureSet `SimulationStructure` R6 class object contain paths of files to be used
+#' @param populationFilePath path to population file to be used for simulation
+#' @param resultsFilePath path to CSV files to which results will be saved,
+#' @param debugLogFileName path to file where core debug logs are saved
+#' @param nodeName node name for parallel simulations
+#' @param showProgress option to print progress of simulation to console
+#' @export
+#' @import ospsuite
+simulateModelOnCore <- function(structureSet,
+                                populationFilePath = NULL,
+                                resultsFilePath,
+                                debugLogFileName = file.path(defaultFileNames$workflowFolderPath(), defaultFileNames$logDebugFile()),
+                                nodeName = NULL) {
+
+  simulation <- ospsuite::loadSimulation(structureSet$simulationSet$simulationFile,
+                                         addToCache = FALSE,
+                                         loadFromCache = FALSE
+  )
+
+  logDebug(message = paste0(ifnotnull(nodeName, paste0(nodeName, ": "), ""),"Simulation file '",structureSet$simulationSet$simulationFile,"' successfully loaded"),
+           file = debugLogFileName,
+           printConsole = FALSE)
+
+  if (!is.null(populationFilePath)) {
+    population <- ospsuite::loadPopulation(populationFilePath)
+    logDebug(
+      message = paste0(ifnotnull(nodeName, paste0(nodeName, ": "), ""),"Population file '",populationFilePath,"' successfully loaded"),
+      file = debugLogFileName,
+      printConsole = FALSE
+    )
+  }
+
+  simRunOptions <- ospsuite::SimulationRunOptions$new(showProgress = FALSE)
+  simulationResult <- ospsuite::runSimulation(simulation, population = population, simulationRunOptions = simRunOptions)
+
+  logDebug(
+    message = paste0(ifnotnull(nodeName, paste0(nodeName, ": "), ""),"Simulation run complete"),
+    file = debugLogFileName,
+    printConsole = FALSE
+  )
+
+  ospsuite::exportResultsToCSV(

```

Comment:
SAme here. Line 84 is one line, line 92 is one parameter per line. I think we need to come up with a consistent way of writing the code. I don't think this is something that we have addressed in the styleguide.

Reply:
fixed